examples/sensors/custom_data_handlers.py

```diff
@@ -0,0 +1,521 @@
+                )
+                continue
+
+            for i, channel_label in enumerate(channel_labels):
+                if i < len(subplot_data):
+                    self.y_data[subplot_key][channel_label].append(float(subplot_data[i]))
+
+        # Safety check: ensure all channels have same length as x_data
+        for subplot_key in self.y_data:
+            for channel_label in self.y_data[subplot_key]:
+                while len(self.y_data[subplot_key][channel_label]) < len(self.x_data):
+                    # Pad with last value or 0 if empty
+                    last_val = (
+                        self.y_data[subplot_key][channel_label][-1] if self.y_data[subplot_key][channel_label] else 0.0
+                    )
+                    self.y_data[subplot_key][channel_label].append(last_val)
+
+        # Maintain rolling window
+        if len(self.x_data) > self.rolling_window_size:
+            self.x_data.pop(0)
+            for subplot_key in self.y_data:
+                for channel_label in self.y_data[subplot_key]:
+                    if len(self.y_data[subplot_key][channel_label]) > 0:
+                        self.y_data[subplot_key][channel_label].pop(0)
+
+        # Update all curves
+        for subplot_key, curves in self.curves.items():
+            channel_labels = list(self.subplot_structure[subplot_key])
+            for curve, channel_label in zip(curves, channel_labels):
+                y_data = self.y_data[subplot_key][channel_label]
+                curve.setData(x=self.x_data, y=y_data)
+
+        if self.app:
+            self.app.processEvents()
+
+    def cleanup(self):
+        """Clean up PyQtGraph resources."""
+        if self.widget:
+            try:
+                self.widget.close()
+                gs.logger.debug("PyQtGraphPlotter: closed PyQtGraph window")
+            except Exception as e:
+                gs.logger.warning(f"PyQtGraphPlotter: Error closing window: {e}")
+            finally:
+                self.widget = None
+                self.plot_widgets.clear()
+                self.curves.clear()
+
+
+class MPLPlotter(DataHandler):
+    """
+    Real-time plot using MatPlotLib for live sensor data visualization.
+
+    Parameters
+    ----------
+    labels : list[str] | dict[str, list[str]] | None
+        Labels for data channels. Can be:
+        - list[str]: Labels for array-like data channels
+        - dict[str, list[str]]: Dict keys become subplot titles, values are channel labels within each subplot
+        - None: Auto-detect from first data input (supports both array and dict data)
+    vis_window_size : tuple[int, int], optional
+        The size of the plot window in inches. Defaults to (10, 6).
+    rolling_window_size : int, optional
+        Number of data points to display in the rolling window. Defaults to 100.
+    title : str, optional
+        Plot window title. Defaults to "Live Sensor Data".
+    """
```

Comment:
There seems to be a lot of duplication between this class and the one above. I would suggest adding some abstract class `DataPlotter`, that would only specialise what is strictly necessary for each framework.

Note that I'm not going to review `MPLPlotter` before you apply these changes.